manifests/service.pp

```diff
@@ -1,58 +1,67 @@
+# Setting up an upstart service
 define upstart::service(
-	$service_name=$title,
-	$ensure=running,
-	$enable=true,
-	$user=undef,
-	$group=undef,
-	$chdir=undef,
-	$chroot=undef,
-	$exec=undef,
-	$script=undef,
-	$expect=undef,
-	$start_on="runlevel [2345]",
-	$stop_on="runlevel [^2345]",
-	$template="upstart/init.conf.erb",
-	$env={},
-	$limits={},
-	$nice=undef,
-        $respawn=false,
-        $respawn_limit=undef,
-        $hooks={}
+  $service_name=$title,
+  $ensure=running,
+  $enable=true,
+  $user=undef,
+  $group=undef,
+  $chdir=undef,
+  $timeout=undef,
+  $chroot=undef,
+  $exec=undef,
+  $script=undef,
+  $expect=undef,
+  $start_on='runlevel [2345]',
+  $stop_on='runlevel [^2345]',
+  $template='upstart/init.conf.erb',
+  $env={},
+  $limits={},
+  $nice=undef,
+  $respawn=false,
+  $respawn_limit=undef,
+  $hooks={}
 ) {
-	include upstart::params
-	include upstart::utils
+  include upstart::params
+  include upstart::utils
 
-        if $respawn_limit!=undef {
-          validate_string($respawn_limit)
-        }
-        validate_bool($respawn)
+  if($respawn_limit!=undef){
+    validate_string($respawn_limit)
+  }
+  validate_bool($respawn)
 
-	file{"${upstart::params::svc_dir}/${service_name}.conf":
-		ensure => $enable ? { true => present, false => absent, default => present},
-		content => template($template),
-		mode => 644,
-		notify => Service[$service_name]
-	}
+  file{"${upstart::params::svc_dir}/${service_name}.conf":
+    content => template($template),
+    mode    => '0644',
+    notify  => Service[$service_name]
+  } ->
 
-	case $::osfamily {
-		"RedHat" : {
-			service { $service_name:
-				ensure => $ensure,
-				hasrestart => true,
-				hasstatus => true,
-				start => "/sbin/initctl start ${service_name}",
-				restart => "/sbin/initctl stop ${service_name}; /sbin/initctl start ${service_name}",
-				stop => "/sbin/initctl stop ${service_name}",
-				status => "/sbin/initctl status ${service_name} | grep running",
-				require => [File["${upstart::params::svc_dir}/${service_name}.conf"],
-					File["/usr/local/sbin/chpst.py"]]
-			}
-		}
-		"Debian" : {
-			service { $service_name:
-				ensure => $ensure,
-				require => File["${upstart::params::svc_dir}/${service_name}.conf"]
-			}
-		}
-	}
+  file{"${upstart::params::svc_dir}.d/${service_name}":
+    ensure => link,
+    target => "${upstart::params::svc_dir}/${service_name}.conf"
```

Comment:
http://upstart.ubuntu.com/cookbook/ clearly states that sysv compatibility
link should be to /lib/init/upstart-job. note this link is not required
from 13.10 onwards

On Tue, Feb 17, 2015 at 1:00 AM, ronen <notifications@github.com> wrote:

> In manifests/service.pp
> <https://github.com/fewbytes-puppet-show/fewbytes-upstart/pull/1#discussion_r24781575>
> :
>
> > -				stop => "/sbin/initctl stop ${service_name}",
> > -				status => "/sbin/initctl status ${service_name} | grep running",
> > -				require => [File["${upstart::params::svc_dir}/${service_name}.conf"],
> > -					File["/usr/local/sbin/chpst.py"]]
> > -			}
> > -		}
> > -		"Debian" : {
> > -			service { $service_name:
> > -				ensure => $ensure,
> > -				require => File["${upstart::params::svc_dir}/${service_name}.conf"]
> > -			}
> > -		}
> > -	}
> > +  file{"${upstart::params::svc_dir}.d/${service_name}":
> > +    ensure => link,
> > +    target => "${upstart::params::svc_dir}/${service_name}.conf"
>
> Not that I'm aware of, this link enables auto completion when using
> service
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fewbytes-puppet-show/fewbytes-upstart/pull/1/files#r24781575>
> .
>